server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -121,6 +117,14 @@ public TemporalAccessor parse(String input) {
         }
     }
 
+    /**
+     * Attempt parsing the input without throwing exception. This is needed because java-time requires ordering on optional (composite)
+     * patterns. Joda does not suffer from this.
+     * https://bugs.openjdk.java.net/browse/JDK-8188771
+     *
+     * @param input An arbitrary string resembling the string representation of a date or time
+     * @return true if parsing was successful, false if parsing failed
```

Comment:
this method is not returning a boolean

Reply:
yes sorry, that was a copypaste from previous rev. Fixed with more description